chronosphere/resource_dashboard.go

```diff
@@ -79,9 +79,14 @@ func (dashboardConverter) toModel(
 		return nil, fmt.Errorf("invalid dashboard_json: %s", err)
 	}
 
-	name := dashboard.Metadata.Name
+	if d.Name != "" && dashboard.Metadata.Name != "" {
+		return nil, errors.New("only one of `name` and `dashboard_json.metadata.name` can be set")
+	}
+
+	// always prefer the top level name explicitly set
+	name := d.Name
 	if name == "" {
-		return nil, errors.New(`invalid dashboard_json: dashboard metadata must have "name" set`)
+		name = dashboard.Metadata.Name
```

Comment:
chronosphere should be the only user of `Metadata` and it's not even supported by the UI anymore. Once we update TF with this internally, we should migrate the dashboards in the CS repo and get rid of the field.

Reply:
yes, but i don't know if you can do that without breaking some dashboards in meta + customer success repos.

Reply:
This is probably the best time to make that change as we cut a release as part of 1.0

Reply:
Bump on this @aschepis @bg451  -- should we remove the "set name from metadata" and require name at the top-level to simplify?